geode-core/src/main/java/org/apache/geode/internal/cache/EntryEventImpl.java

```diff
@@ -189,6 +189,114 @@ public EntryEventImpl() {
     // do nothing
   }
 
+  /** create a callback event for applying a transactional change to the local cache */
+  @Retained
+  public static EntryEventImpl createCallbackEvent(final InternalRegion internalRegion,
```

Comment:
It might be better to move this to an EntryEventFactory type class instead of EntryEventImpl.

Reply:
We were hesitant at first to introduce more classes. But separating this functionallity into its own class is probably makes its intent clearer. Will change